File: 백준/Silver/14425. 문자열 집합/문자열 집합.py
import sys

n,m = (map(int,sys.stdin.readline().split()))

arr = []
for i in range(n):
    arr.append(sys.stdin.readline())
arr = set(arr)

count = 0
for i in range(m):
    if sys.stdin.readline() in arr:
        count += 1
print(count)

File: 백준/Silver/14425. 문자열 집합/문자열 집합.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sb = StringBuilder()

    val (n, m) = br.readLine().split(" ").map { it.toInt() }
    val words = mutableSetOf<String>()
    for (i in 0 until n) {
        words.add(br.readLine())
    }

    var count = 0
    for (i in 0 until m) {
        val word = br.readLine()
        if (words.contains(word)) {
            count++
        }
    }
    sb.append(count)

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}